torch/_inductor/debug.py

```diff
@@ -387,3 +391,66 @@ def graph_diagram(self, nodes: SchedulerNodeList):
 
     def output_code(self, filename):
         shutil.copy(filename, self.filename("output_code.py"))
+
+
+save_args_cnt = itertools.count()
+
+
+def save_args(fn):
+    if not config.save_args:
```

Comment:
Thanks for catching this subtle issue!

Using decorator has the benefit that we can reliably get argument lists for the decorated function. This will work without code change even if people add/remove arguments.

To resolve the issue you mentioned, maybe I could also move the check inside the wrapper on L404.